update state and add useRef

codercreative opened the pull request.

JoHeyJo commented:
Be sure to add descriptive PR messages. Consider the assignment requirements and use that as a guide.

JoHeyJo commented on src/App.jsx:
```diff
@@ -4,13 +4,20 @@ import TodoForm from './TodoForm.jsx';
+  const [todoList, setTodoList] = useState([]);
+
+  function addTodo(title) {
+    // using Date.now() to create a separate and distinct key for all the todos
```
let's clean up these comments. Reserve comments for essential information that is not clearly conveyed from the code.

codercreative commented:
Note: Removed non-essential comments from the lesson branch as requested and pushed the updated code.

====================

PR Description:

In App component: Refactor the useState hook and use an empty array as the initial value.

In App component: Create a handler function that takes in the todo title and uses state to list the previous todos with the new todos.

In App component: Within the TodoForm instance, I pass the addTodo function to an onAddTodo props.

In TodoForm component: In the input tag, I added a name props with the value of "title".

In TodoForm component: I added a handleAddTodo to handle the form event—preventing the form from automatically submitting, saving the user input in a const, deleting the old user input to make room for user's next todo, and maintaining focus state on the input tag (by using the useRef hook) for better user experience after user hits enter.

In App component: I add the todoList state value to the TodoList instance.

In TodoList component: I destructure the todoList out of component's props argument and use todoList to map over every todo to display every todo item, each todo logging a unique key generated with the help of Date.now().
